src/mem/freelist.h

```diff
@@ -121,11 +93,42 @@ namespace snmalloc
     }
 
     /**
-     * Read the next pointer handling any required decoding of the pointer
+     * Static so that it can be used on reference to a FreeObject.
```

Comment:
```suggestion
     * Assign next_object and update its prev_encoded if CHECK_CLIENT.
     *
     * Static so that it can be used on reference to a FreeObject.
```